lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -6703,59 +6769,47 @@ DefConstructor('\@PADDED[MuDimension]{MuDimension}{}',
-DefMathI('\@INVISIBLEPLUS', undef, "\x{2064}", reversion => '', name => '', meaning => 'plus', role => 'ADDOP');
-
-# We OUGHT to be able to do this using \llap,\rlap,\hss...
-DefMacro('\lx@tweaked{}{}', '\ifmmode\lx@math@tweaked{#1}{#2}\else\lx@text@tweaked{#1}{#2}\fi');
-DefConstructor('\lx@math@tweaked RequiredKeyVals {}',
-  "<ltx:XMWrap %&GetKeyVals(#1)>#2</ltx:XMWrap>",
-  afterDigest => sub {
-    my ($stomach, $whatsit) = @_;
-    my ($kv,      $body)    = $whatsit->getArgs;
-    $whatsit->setProperties($kv->getPairs);
-    $whatsit->setFont($body->getFont);
+#======================================================================
+# Building XMDuals for Mathematical Parallel markup
+# Used when the content and presentation forms have different structure.
+
+DefKeyVal('XMath', 'reversion',              'UndigestedKey');
+DefKeyVal('XMath', 'content_reversion',      'UndigestedKey');
+DefKeyVal('XMath', 'presentation_reversion', 'UndigestedKey');
+our @PENDING_DUAL_XMARGS = ();
+DefConstructor('\lx@dual OptionalKeyVals:XMath {}{}',
+  #  "<ltx:XMDual role='&GetKeyVal(#1,role)'>#2<ltx:XMWrap>#3</ltx:XMWrap></ltx:XMDual>",
+  "<ltx:XMDual role='#role'>#2<ltx:XMWrap>#3</ltx:XMWrap></ltx:XMDual>",
+  beforeDigest => sub {
+    push(@PENDING_DUAL_XMARGS, {});
     return; },
-  reversion => '#2');
-
-DefConstructor('\lx@text@tweaked RequiredKeyVals {}',
-  "<ltx:text _noautoclose='1' %&GetKeyVals(#1)>#2</ltx:text>",
   afterDigest => sub {
     my ($stomach, $whatsit) = @_;
-    my ($kv,      $body)    = $whatsit->getArgs;
-    $whatsit->setProperties($kv->getPairs); });
-
-DefMacro('\lx@nounicode {}', '\ifmmode\lx@math@nounicode#1\else\lx@text@nounicode#1\fi');
-
-DefConstructor('\lx@framed[]{}',
-  "<ltx:text framed='#frame' _noautoclose='1'>#2</ltx:text>",
-  properties => { frame => sub { ToString($_[1] || 'rectangle'); } });
-DefConstructor('\lx@hflipped{}',
-  "<ltx:text class='ltx_hflipped' _noautoclose='1'>#1</ltx:text>");
-
-sub reportNoUnicode {
-  my ($cs) = @_;
-  $cs = ToString($cs);
-  if (!LookupMapping('missing_unicode' => $cs)) {
-    Warn('expected', 'unicode', $cs,
-      "There's no Unicode equivalent for the symbol '$cs'");
-    AssignMapping('missing_unicode' => $cs => 1); }
-  return; }
-# Slightly contrived so that this can be used within a DefMath
-# and still declare & get the semantic properties.
-DefPrimitive('\lx@math@nounicode DefToken', sub {
-    my ($stomach, $cs) = @_;
-    reportNoUnicode($cs);
-    Box(ToString($cs), undef, undef, $cs, class => 'ltx_nounicode'); });
-
-DefConstructor('\lx@text@nounicode DefToken',
-  "<ltx:text _no_autoclose='true' class='ltx_nounicode'>#1</ltx:text>",
-  afterDigest => sub {
-    reportNoUnicode(ToString($_[1]->getArg(0))); });
+    my $kv     = $whatsit->getArg(1);
+    my $xmargs = pop(@PENDING_DUAL_XMARGS);
+    $whatsit->setProperties(%$xmargs);    # Hopefully no name class with XM<digits>
```

Comment:
this would be a perl "die" if $xmargs came back undefined from the pop() call right? Maybe an `if $xmargs`...

Reply:
Yep, hopefully coded a bit more robustly now.